File: jdroid-android-about/src/main/java/com/jdroid/android/about/AboutRemoteConfigParameter.java
package com.jdroid.android.about;

import com.jdroid.android.firebase.remoteconfig.RemoteConfigParameter;

public enum AboutRemoteConfigParameter implements RemoteConfigParameter {
	
	RATE_APP_MIN_DAYS_SINCE_LAST_RESPONSE(90),
	RATE_APP_MIN_DAYS_SINCE_FIRST_APP_LOAD(7),
	RATE_APP_MIN_APP_LOADS(10),
	RATE_APP_MIN_DAYS_SINCE_LAST_CRASH(21);

	private Object defaultValue;

	AboutRemoteConfigParameter() {
		this(null);
	}

	AboutRemoteConfigParameter(Object defaultValue) {
		this.defaultValue = defaultValue;
	}

	@Override
	public String getKey() {
		return name();
	}

	@Override
	public Object getDefaultValue() {
		return AboutAppModule.get().getAboutContext().getBuildConfigValue(name(), defaultValue);
	}

	@Override
	public Boolean isUserProperty() {
		return false;
	}
}

File: jdroid-android-about/src/main/java/com/jdroid/android/about/AboutRemoteConfigParameter.kt
package com.jdroid.android.about

import com.jdroid.java.remoteconfig.RemoteConfigParameter

enum class AboutRemoteConfigParameter constructor(private val defaultValue: Any? = null) : RemoteConfigParameter {

    RATE_APP_MIN_DAYS_SINCE_LAST_RESPONSE(90),
    RATE_APP_MIN_DAYS_SINCE_FIRST_APP_LOAD(7),
    RATE_APP_MIN_APP_LOADS(10),
    RATE_APP_MIN_DAYS_SINCE_LAST_CRASH(21);

    override fun getKey(): String {
        return name
    }

    override fun getDefaultValue(): Any? {
        return AboutAppModule.get().aboutContext.getBuildConfigValue<Any>(name, defaultValue)
    }
}
